### ISSUE TITLE
Please make attributes visible in item lists

### ISSUE THREAD

**Issue by haeckse** [opensourcepos/opensourcepos#2890]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.1
- OSPOS git commit hash is: c786d4
- PHP version is:  PHP 7.4
- MySQL or MariaDB version is: 5.7
- OS and version is: -
- WebServer is: 1and1 webhost
- Selected language is: en/de

### New Feature

In the settings one can currently set the type for 3 columns that are shown in the article list in sales mode. One can choose from 'article name', 'article number' and 'price'. If one has a lot of products of the same type, but with a number of variations, it would be nice if the attributes would show up in the list.

Here's an example for using POS in a Pizzeria:

![Bildschirmfoto 2020-07-13 um 11 06 33](https://user-images.githubusercontent.com/50916833/87286288-f8017500-c4f8-11ea-85e3-04d7193e43fb.png)

As a workaround, to be able to determine the different Pizza sizes, the size has been added to the 'article number' column to make it show up in the search result. 

It would be even nicer if attributes like 'size' or 'additional topping' would be selectable after one has choosen a pizza (or after one has added it to the shopping cart).

**Comment by jekkos** [comment]
Ok this makes sense. I guess that if you select it as visible in the sales register then you should also see the value in this dropdown.

**Comment by objecttothis** [comment]
@jekkos agreed that this is a good idea.  The settings dropdowns would need to be extended in the view to include attribute names and then the business logic in the sales register would need to be modified to look up either the attribute or the items table. Very doable.  I'm guessing this is a 3.4.0 change.  Hopefully we can release 3.2.2 soon so we can attend to those things (wink wink).

**Comment by rocky8roy** [comment]
@haeckse Hi
if you want this feature kindly contact me on rocky8roy@gmail.com to discuss further
thank you

**Comment by haeckse** [comment]
@jekkos  I want to inform you, that I have been asked to pay for the implementation of this feature request. I have filed this issue because I consider OSPOS to be a classic open-source project, and I saw ways for enhancements when installing OSPOS on a server for a friend's Pizzeria. I myself do not own a business, so I would not benefit from such enhancements. 
If paying for new features is an intended option, you may consider to mention it in the “background information” text that one gets presented with when one opens a new issue or feature request.

Anyway, I add a document here, which I have compiled when discussing the implementation. I think it makes the request more clear. I have also described a more sophisticated solution for this feature, which I find worth considering.

[ospos_fr_2890.pdf](https://github.com/opensourcepos/opensourcepos/files/5573621/ospos_fr_2890.pdf)

**Comment by objecttothis** [comment]
> 
> 
> @jekkos I want to inform you, that I have been asked to pay for the implementation of this feature request. I have filed this issue because I consider OSPOS to be a classic open-source project, and I saw ways for enhancements when installing OSPOS on a server for a friend's Pizzeria. I myself do not own a business, so I would not benefit from such enhancements.
> If paying for new features is an intended option, you may consider to mention it in the “background information” text that one gets presented with when one opens a new issue or feature request.
> 
> Anyway, I add a document here, which I have compiled when discussing the implementation. I think it makes the request more clear. I have also described a more sophisticated solution for this feature, which I find worth considering.
> 
> [ospos_fr_2890.pdf](https://github.com/opensourcepos/opensourcepos/files/5573621/ospos_fr_2890.pdf)

@haeckse Because of the nature of the project, OSPOS can be contributed to by anyone with the skillset required to do that.  People submit feature requests all the time and often the ones that get implemented are the ones that the developers themselves benefit from.  For example, I run a business that may benefit from this feature, so I don't mind submitting the Pull Request with the code for free.  Other feature requests that would benefit the whole community but just aren't that high on contributors lists may not get implemented.  It is perfectly acceptable for a legitimate freelance dev to offer to implement the feature and if you need it enough/price is right for you then you can pay them to do the work.

Keep in mind that the dev would still need to submit the Pull Request and it still needs to pass code review by other contributors if you want the enhancement implemented into the OSPOS project.  If not, the dev can clone it and give you a cloned copy and then it doesn't need to pass all that, but it will be stuck at that version of the code unless you or someone keeps it sync'd with the OSPOS repository.

**Comment by objecttothis** [comment]
@haeckse to clarify, I would benefit from this feature but it isn't high on my list of features that I need anytime soon, so it would take me awhile to get to it.  My comment above was not meant as a promise to do the work.

**Comment by haeckse** [comment]
@objecttothis Thanks for the clarification. Of course, any developer can offer paid services as a contractor for making amendments if it is permitted by the software license. But the intention should be clearly communicated from the very beginning, not after having emailed back and forth a couple of days. Maybe I got a bit too upset about it, but anyway, as mentioned, I personally do not need the feature, so just take it as a suggestion. Also, the guys from the pizzeria are happy with the workaround as is. So, take your time :)

**Comment by objecttothis** [comment]
I see.  It sounds like the primary issue is that you thought that @rocky8roy offered to help with the enhancement for free but it was only after several exchanges that you realized that he was wanting to do the enhancement for a fee.  Sounds like a misunderstanding.  I don't think there is anything in the license that prevents people from charging for code contributions.  It's primarily there so that people don't claim the work as their own.

**Comment by rocky8roy** [comment]
> > @jekkos I want to inform you, that I have been asked to pay for the implementation of this feature request. I have filed this issue because I consider OSPOS to be a classic open-source project, and I saw ways for enhancements when installing OSPOS on a server for a friend's Pizzeria. I myself do not own a business, so I would not benefit from such enhancements.
> > If paying for new features is an intended option, you may consider to mention it in the “background information” text that one gets presented with when one opens a new issue or feature request.
> > Anyway, I add a document here, which I have compiled when discussing the implementation. I think it makes the request more clear. I have also described a more sophisticated solution for this feature, which I find worth considering.
> > [ospos_fr_2890.pdf](https://github.com/opensourcepos/opensourcepos/files/5573621/ospos_fr_2890.pdf)
> 
> @haeckse Because of the nature of the project, OSPOS can be contributed to by anyone with the skillset required to do that. People submit feature requests all the time and often the ones that get implemented are the ones that the developers themselves benefit from. For example, I run a business that may benefit from this feature, so I don't mind submitting the Pull Request with the code for free. Other feature requests that would benefit the whole community but just aren't that high on contributors lists may not get implemented. It is perfectly acceptable for a legitimate freelance dev to offer to implement the feature and if you need it enough/price is right for you then you can pay them to do the work.
> 
> Programmers need to eat, so I would recommend seeing it less like a mercenary and more like hiring a contractor to take out a wall in your house.
> 
> Keep in mind that the dev would still need to submit the Pull Request and it still needs to pass code review by other contributors if you want the enhancement implemented into the OSPOS project. If not, the dev can clone it and give you a cloned copy and then it doesn't need to pass all that, but it will be stuck at that version of the code unless you or someone keeps it sync'd with the OSPOS repository.

thanks for your explanation!
just in case if anyone want this feature I can do it for a nominal fee and would love to PR too if client agrees
thank you!!!!!!

**Comment by objecttothis** [comment]
> thanks for your explanation!
> just in case if anyone want this feature I can do it for a nominal fee and would love to PR too if client agrees
> thank you!!!!!!

@rocky8roy If someone wants to do this, it's fine, but so that there is no misunderstanding, please make sure that you specify at the beginning of your conversation that your work is paid.
